### ISSUE TITLE
Allow us to set how uploaded data should be interpreted.

### ISSUE THREAD

**Issue by Ctilhs** [maybelinot/df2gspread#44]
From what I understand, by default when we use the upload() from df2gspread, dataframe values are converted to string then upload is performed.

I struggled with it when I had to upload values which were prices. Same situation if I would like them to stay as digits after update, without my interaction in google docs.

Simple fix.
In upload() add argument ```value_input_opt='RAW'```
https://github.com/maybelinot/df2gspread/blob/01bc46d267c8ea42098f425758930bb28cab0573/df2gspread/df2gspread.py#L24-L26

And in three function calls ```value_input_option=value_input_opt```
https://github.com/maybelinot/df2gspread/blob/01bc46d267c8ea42098f425758930bb28cab0573/df2gspread/df2gspread.py#L126
https://github.com/maybelinot/df2gspread/blob/01bc46d267c8ea42098f425758930bb28cab0573/df2gspread/df2gspread.py#L134
https://github.com/maybelinot/df2gspread/blob/01bc46d267c8ea42098f425758930bb28cab0573/df2gspread/df2gspread.py#L147

From: [developers.google.com](https://developers.google.com/sheets/api/reference/rest/v4/ValueInputOption)
What we get? Now in upload we could set parameter ```value_input_option='USER_ENTERED'```, so when we upload digits they will be automatically parsed as numbers. 
If we will not specify ```value_input_option``` in upload() it will stay default as ```'RAW'``` so data will be uploaded as it is now.

Not sure is it related?: https://github.com/maybelinot/df2gspread/issues/41

It's my first time ever adding Issue at github. I will gladly accept advice if I should improve something.

**Comment by saitx** [comment]
Yeah for now I have been having a duplicate sheet with =VALUE(Raw_Sheet!H2) and so on. This would be a great addition.

**Comment by maybelinot** [comment]
@Ctilhs thank you for creating an issue, that would be great if you could submit a PR with proposed changes. Otherwise I'll work on this as soon as I'll have time.

**Comment by winf-hsos** [comment]
Any updates on this?

**Comment by varunjain3** [comment]
Are the authors alive? I wanted this as well, shall I create a pr?

**Comment by maybelinot** [comment]
Hi @varunjain3 I don't actively support the lib anymore, if you will open a pr with tests - I'll review and bump the lib version

**Comment by varunjain3** [comment]
Thanks for the prompt reply, I’ll do the needful, also are there any alternatives you found for the same project?

**Comment by maybelinot** [comment]
Not really, df2gspread mainly leverages https://github.com/burnash/gspread, which is update regularly from what I can see, so you may built small abstraction layer to fulfil your needs just on top of that.

**Comment by mroy-seedbox** [comment]
I'd suggest abandoning (that's what we did), when an entire library can be replaced with a couple lines of monkey patching in the actively maintained [`gspread`](https://docs.gspread.org/en/latest/) lib:
```
    def upload_pandas_df(self, df):
      values = [df.columns.values.tolist()]
      values.extend(df.values.tolist())
      sheet.values_update(
        self.title,
        params = { 'valueInputOption': 'USER_ENTERED' },
        body = { 'values': values }
      )

    gspread.Worksheet.upload_pandas_df = upload_pandas_df
```

Or use [gspread-dataframe](https://github.com/robin900/gspread-dataframe) instead.
